dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/HadoopScripts.java

```diff
@@ -31,13 +62,16 @@ static byte[] read(String scriptFilename) throws IOException {
   }
 
   /** Extracts the script from the resources directory inside the jar to the local filesystem. */
-  static File extract(String scriptFilename) throws IOException {
+  static synchronized Path extract(String scriptFilename) throws IOException {
```

Comment:
why the `synchronized` ?

Reply:
It is a left-over from the previous version of the code. Since memoized supplier is synchronized now, there is no need for this additional synchronization. Fixed.